api/controller.py

```diff
@@ -165,12 +180,17 @@ def __init__(self, _db, lanes=None, testing=False):
             self.circulation_apis[library.id] = self.setup_circulation(
                 library
             )
-            self.setup_adobe_vendor_id(library)
+            authdata = self.setup_adobe_vendor_id(library)
+            if authdata and not self.adobe_device_management:
+                # There's at least one library on this system that
+                # wants Vendor IDs. This means we need to advertise support
+                # for the Device Management Protocol.
+                self.adobe_device_management = DeviceManagementProtocolController(self)
```

Comment:
Previously, `self.adobe_device_management` was set within `setup_adobe_vendor_id`. This wasn't the best because it meant the controller could be set up multiple times if multiple libraries on the system had Short Client Token integrations.